src/main/java/com/igot/cb/util/PayloadValidation.java

```diff
@@ -45,9 +45,16 @@ public String validateAccessControlPayload(Map<String, Object> payload) {
                 if (criteriaObj instanceof Map) {
                   Map<String, Object> criteria = (Map<String, Object>) criteriaObj;
                   Object criteriaValueObj = criteria.get(Constants.CRITERIA_VALUE);
-                  if (!(criteriaValueObj instanceof List) || ((List<?>) criteriaValueObj).isEmpty()) {
-                    errList.add("userGroups[" + i + "].userGroupCriteriaList[" + j + "].criteriaValue");
-                  }
+                    if (criteriaValueObj == null) {
+                        errList.add("userGroups[" + i + "].userGroupCriteriaList[" + j + "].criteriaValue");
+                    } else if (criteriaValueObj instanceof List) {
+                        if (((List<?>) criteriaValueObj).isEmpty()) {
+                            errList.add("userGroups[" + i + "].userGroupCriteriaList[" + j + "].criteriaValue");
+                        }
+                    } else if (!(criteriaValueObj instanceof Boolean) && !(criteriaValueObj instanceof String)) {
+
+                        errList.add("userGroups[" + i + "].userGroupCriteriaList[" + j + "].criteriaValue");
+                    }
```

Comment:
you are adding same error to errList in 3 different places... 
create a boolean isError and set to false.. 

if value is null or not as expected type - set isError to true
at the end check isError if true update the errList object